index.js

```diff
@@ -1,65 +1,257 @@
+    /** Disallow unreachable code after `return`, `throw`, `continue`, and `break` statements */
+    'no-unreachable': 'error',
+    /** Disallow control flow statements in `finally` blocks */
+    'no-unsafe-finally': 'error',
+    /** Disallow negating the left operand of relational operators */
+    'no-unsafe-negation': 'error',
+    /** Require calls to `isNaN()` when checking for `NaN` */
+    'use-isnan': 'error',
+    /** Enforce comparing `typeof` expressions against valid strings */
+    'valid-typeof': 'error',
+
+    /**
+     * Best Practices
+     */
+    /** Enforce `return` statements in callbacks of array methods */
+    'array-callback-return': ['error', { allowImplicit: true }],
+    eqeqeq: ['error', 'always', { null: 'ignore' }],
+    /** Require `for-in` loops to include an `if` statement */
     'guard-for-in': 'error',
+    /** Disallow `else` blocks after `return` statements in `if` statements */
     'no-else-return': 'error',
+    /** Disallow empty destructuring patterns */
+    'no-empty-pattern': 'error',
+    /** Disallow the use of `eval()` */
     'no-eval': 'error',
+    /** Disallow extending native types */
     'no-extend-native': 'error',
+    /** Disallow fallthrough of `case` statements */
+    'no-fallthrough': 'error',
+    /** Disallow assignments to native objects or read-only global variables */
+    'no-global-assign': 'error',
+    /** Disallow the use of `eval()`-like methods */
     'no-implied-eval': 'error',
+    /** Disallow `new` operators outside of assignments or comparisons */
     'no-new': 'error',
+    /** Disallow `new` operators with the `Function` object */
     'no-new-func': 'error',
+    /** Disallow `new` operators with the `String`, `Number`, and `Boolean` objects */
     'no-new-wrappers': 'error',
+    /** Disallow octal literals */
+    'no-octal': 'error',
+    /** Disallow variable redeclaration */
+    'no-redeclare': 'error',
+    /** Disallow assignments where both sides are exactly the same */
+    'no-self-assign': 'error',
+    /** Disallow `javascript:` urls */
     'no-script-url': 'error',
+    /** Disallow throwing literals as exceptions */
     'no-throw-literal': 'error',
+    /** Disallow unused labels */
+    'no-unused-labels': 'error',
+    /** Disallow unnecessary `catch` clauses */
+    'no-useless-catch': 'error',
+    /** Disallow unnecessary escape characters */
+    'no-useless-escape': 'error',
+    /** Disallow `with` statements */
+    'no-with': 'error',
+
+    /**
+     * Variables
+     */
+    /** Disallow deleting variables */
+    'no-delete-var': 'error',
+    /** Disallow variable declarations from shadowing variables declared in the outer scope */
     'no-shadow': 'error',
+    /** Disallow identifiers from shadowing restricted names */
+    'no-shadow-restricted-names': 'error',
+    /** disallow the use of undeclared variables unless mentioned in global comments */
+    'no-undef': 'error',
+
+    /**
+     * Stylistic Issues
+     */
```

Comment:
I wrote in the intro we shouldn't have stylistic rules :)

Do we really need both of these? Prettier should take care of spaces/tabs mix, and I usually protest these "prefer-" rules, as often they do more harm than good. If `Object.assign` is used, it's most often more than just a spread that is needed, since you're often dealing with larger pieces of logic that aren't using modern syntax.

Reply:
Removed.